rust/private/rust.bzl

```diff
@@ -119,7 +119,7 @@ def get_edition(attr, toolchain, label):
     """
     if getattr(attr, "edition"):
         return attr.edition
-    elif not toolchain.default_edition:
+    elif toolchain.default_edition == "required" or not toolchain.default_edition:
```

Comment:
I think the order of comparisons would need to be switched to avoid an error (None check first).